TestingHelpers/MockFileVersionInfo.cs

```diff
@@ -0,0 +1,193 @@
+﻿namespace System.IO.Abstractions.TestingHelpers
+{
+    [Serializable]
+    public class MockFileVersionInfo : MockFileVersionInfoMutableAdapter
```

Comment:
TBH, I don't really like this ceremony just to have setters for the properties.

What about keeping everything immutable and allowing users to pass the necessary values in `MockFileVersionInfo`s constructor either as optional parameters or via a parameter class with setters?